### PR TITLE
Accessibility extension   FR49

### PR DESCRIPTION
**Pull request by Aurige**
Addition of equipment's attributes for Accessibility

### CODE REVIEW

File: xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentSigns_version.xsd [hunk 1]
```diff
@@ -158,6 +158,21 @@ Rail transport, Roads and Road transport
...
+			<xsd:element name="AudioAnnoucementType" type="AudioAnnoucementTypeEnumeration" minOccurs="0">
+				<xsd:annotation>
+					<xsd:documentation>Indicate whether the display may be read by the audio tool .</xsd:documentation>
+				</xsd:annotation>
+			</xsd:element>
+			<xsd:element name="FontSize" type="FontSizeEnumeration" minOccurs="0">
+				<xsd:annotation>
+					<xsd:documentation>Character font size .</xsd:documentation>
+				</xsd:annotation>
+			</xsd:element>
```
Comment (skinkie) [inline xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentSigns_version.xsd:170]: @nick-knowles in your version FontSize is not present. Please advise.

File: xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentSigns_version.xsd [hunk 2]
```diff
@@ -328,6 +343,32 @@ Rail transport, Roads and Road transport
 			<xsd:enumeration value="other"/>
 		</xsd:restriction>
 	</xsd:simpleType>
+
+	<xsd:simpleType name="AudioAnnoucementTypeEnumeration">
```
Comment (skinkie) [inline xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentSigns_version.xsd:347]: @nick-knowles @Aurige  I am seeing completely different values in master, while @nick-knowles 's famous modification changes suggest they should be there.

```xml
        <xsd:simpleType name="AudioAnnouncementTypeEnumeration">
                <xsd:annotation>
                        <xsd:documentation>Allowed values for AudioAnnouncements </xsd:documentation>
                </xsd:annotation>
                <xsd:restriction base="xsd:NMTOKEN">
                        <xsd:enumeration value="onDemand"/>
                        <xsd:enumeration value="automatic"/>
                </xsd:restriction>
        </xsd:simpleType>
```

File: xsd/netex_part_1/part1_ifopt/netex_ifopt_path_version.xsd [hunk 3]
```diff
@@ -111,6 +111,47 @@ Rail transport, Roads and Road transport
...
+				<xsd:annotation>
+					<xsd:documentation>Type of border on the right side ((in the direction of the pathLink way).</xsd:documentation>
+				</xsd:annotation>
+			</xsd:element>
+			<xsd:element name="LeftSideBorder" type="BorderTypeEnumeration" minOccurs="0">
+				<xsd:annotation>
+					<xsd:documentation>Type of border on the left side ((in the direction of the pathLink way).</xsd:documentation>
+				</xsd:annotation>
+			</xsd:element>			
+			<xsd:element name="TiltAngle" type="xsd:integer" minOccurs="0">
```
Comment (skinkie) [inline xsd/netex_part_1/part1_ifopt/netex_ifopt_path_version.xsd:135]: @nick-knowles I notice that neighter TiltAngle, CodedTilt are part of the current master, while TiltAngle seem to appear in PathNavigationGroup. Please advise.

**Comment by skinkie** [comment]
As you can see from this comment, this is too big of a change. 

Lacks documentation: NecessaryForceEnumeration, SafeForGuideDog, AudioOrVideoIntercom, TactileWarningStrip, AudioAnnouncementType, AudioAnnouncementsTrigger, LockingTypeEnumeration, PathDescriptionGroup (FlooringType, etc.)

image

image



Needs update: StairGroup (WithoutRiser)
image

Wrongly documented: TravelatorEquipment (EnergySaving, while Integrate**s**AnEscalatorPart is defined)

image

Typo un**k**nown
image


Documented:
image

image

image

image

image

image

image

image

image

image

image

image

image
image

image

image

image

image

image

image
image

File: xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentSigns_version.xsd (hunk 1 above)
Comment (Aurige) [reply xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentSigns_version.xsd:170]: I confirm that FontSize is important

File: xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentSigns_version.xsd (hunk 2 above)
Comment (Aurige) [reply xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentSigns_version.xsd:347]: yes, the one from the PR is the one agreed (should be in the doc) and is more comprehensive

**Review by Aurige — COMMENTED** [review]
@nick-knowles I gues we need to work together on that one

File: xsd/netex_part_1/part1_ifopt/netex_ifopt_path_version.xsd (hunk 3 above)
Comment (opyh) [reply xsd/netex_part_1/part1_ifopt/netex_ifopt_path_version.xsd:135]: Shouldn't this be called `Slope`?

Comment (skinkie) [reply xsd/netex_part_1/part1_ifopt/netex_ifopt_path_version.xsd:135]: @nick-knowles @Aurige what does transmodel state?

Comment (Aurige) [reply xsd/netex_part_1/part1_ifopt/netex_ifopt_path_version.xsd:135]: Transmodel doesn't (yet?) go to that level of details on all attributes related to accessibility. But we definitely need the Tilt information on NavigationPath

Comment (opyh) [reply xsd/netex_part_1/part1_ifopt/netex_ifopt_path_version.xsd:135]: I hope this shameless plug is appropriate: If you or somebody you know needs consulting with best practices regarding accessibility criteria, let me know, we have been working on this topic for more than a decade.

We're Sozialhelden e.V., a non-profit, building [Wheelmap.org](https://wheelmap.org), [A11yJSON](https://a11yjson.org),  and founded the [W3C LDA CG](https://www.w3.org/groups/cg/lda).

As the number of accessibility criteria like this is going to grow in data formats, there might be hundreds of properties like this soon, which means some kind of alignment will help everyone involved. Just sign up for the W3C LDA CG, and if you need specific consulting for your specs, send us a mail to developers@sozialhelden.de, we're happy to get involved and there might be some things that you find useful :)

(CC @holgerd)

Comment (skinkie) [reply xsd/netex_part_1/part1_ifopt/netex_ifopt_path_version.xsd:135]: @opyh are you already participating in the actual discussion on basecamp?

Comment (opyh) [reply xsd/netex_part_1/part1_ifopt/netex_ifopt_path_version.xsd:135]: I'll send you a PM. Thanks!

File: xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentWaiting_version.xsd [hunk 4]
```diff
@@ -161,6 +161,21 @@ Rail transport, Roads and Road transport
 					<xsd:documentation>Type of Locker.</xsd:documentation>
 				</xsd:annotation>
 			</xsd:element>
+			<xsd:element name="WheelcahairAccepted" type="xsd:boolean" minOccurs="0">
+				<xsd:annotation>
+					<xsd:documentation>Indicates whether a wheelchair may be stored in thr locker</xsd:documentation>
```
Comment (skinkie) [inline xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentWaiting_version.xsd:166]:
```suggestion
					<xsd:documentation>Indicates whether a wheelchair may be stored in the locker</xsd:documentation>
```

File: xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentWaiting_version.xsd [hunk 5]
```diff
@@ -161,6 +161,21 @@ Rail transport, Roads and Road transport
 					<xsd:documentation>Type of Locker.</xsd:documentation>
 				</xsd:annotation>
 			</xsd:element>
+			<xsd:element name="WheelcahairAccepted" type="xsd:boolean" minOccurs="0">
```
Comment (skinkie) [inline xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentWaiting_version.xsd:164]:
```suggestion
			<xsd:element name="WheelchairAccepted" type="xsd:boolean" minOccurs="0">
```

File: xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentSigns_version.xsd [hunk 6]
```diff
@@ -158,6 +158,21 @@ Rail transport, Roads and Road transport
...
 				</xsd:annotation>
 			</xsd:element>
+			<xsd:element name="AudioAnnoucementType" type="AudioAnnoucementTypeEnumeration" minOccurs="0">
+				<xsd:annotation>
+					<xsd:documentation>Indicate whether the display may be read by the audio tool .</xsd:documentation>
+				</xsd:annotation>
+			</xsd:element>
+			<xsd:element name="FontSize" type="FontSizeEnumeration" minOccurs="0">
+				<xsd:annotation>
+					<xsd:documentation>Character font size .</xsd:documentation>
```
Comment (skinkie) [inline xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentSigns_version.xsd:168]:
```suggestion
					<xsd:documentation>Character font size.</xsd:documentation>
```

File: xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentSigns_version.xsd [hunk 7]
```diff
@@ -328,6 +343,32 @@ Rail transport, Roads and Road transport
...
+	<xsd:simpleType name="FontSizeEnumeration">
+		<xsd:annotation>
+			<xsd:documentation>Allowed values for size of font.</xsd:documentation>
+		</xsd:annotation>
+		<xsd:restriction base="xsd:string">
+			<xsd:enumeration value="verySmall"/>
+			<xsd:enumeration value="Small"/>
+			<xsd:enumeration value="Meduim"/>
+			<xsd:enumeration value="Big"/>
+			<xsd:enumeration value="xxxxx"/>
```
Comment (skinkie) [inline xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentSigns_version.xsd:367]: Why xxxxx?

File: xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentSigns_version.xsd [hunk 8]
```diff
@@ -328,6 +343,32 @@ Rail transport, Roads and Road transport
...
+		</xsd:restriction>
+	</xsd:simpleType>
+	<xsd:simpleType name="FontSizeEnumeration">
+		<xsd:annotation>
+			<xsd:documentation>Allowed values for size of font.</xsd:documentation>
+		</xsd:annotation>
+		<xsd:restriction base="xsd:string">
+			<xsd:enumeration value="verySmall"/>
+			<xsd:enumeration value="Small"/>
+			<xsd:enumeration value="Meduim"/>
```
Comment (skinkie) [inline xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentSigns_version.xsd:365]:
```suggestion
			<xsd:enumeration value="Medium"/>
```

File: xsd/netex_part_1/part1_ifopt/netex_ifopt_localService_version.xsd [hunk 9]
```diff
@@ -381,7 +381,12 @@ Rail transport, Roads and Road transport
 					<xsd:documentation>Depth of Locker.</xsd:documentation>
 				</xsd:annotation>
 			</xsd:element>
-		</xsd:sequence>
+			<xsd:element name="LuggageMaximalWeigth" type="xsd:decimal" minOccurs="0">
+				<xsd:annotation>
+					<xsd:documentation>Maximum weight of the luggage (in kilograms).</xsd:documentation>
+				</xsd:annotation>
+			</xsd:element>
+			</xsd:sequence>
```
Comment (skinkie) [inline xsd/netex_part_1/part1_ifopt/netex_ifopt_localService_version.xsd:389]:
```suggestion
		</xsd:sequence>
```

File: xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentAccess_version.xsd [hunk 10]
```diff
@@ -650,6 +671,21 @@ Rail transport, Roads and Road transport
 					<xsd:documentation>Speed of travelator.</xsd:documentation>
 				</xsd:annotation>
 			</xsd:element>
+			<xsd:element name="Length" type="LengthType" minOccurs="0">
+				<xsd:annotation>
+					<xsd:documentation>Length (integer in meters).</xsd:documentation>
+				</xsd:annotation>
+			</xsd:element>
+			<xsd:element name="Slope" type="xsd:integer" minOccurs="0">
```
Comment (skinkie) [inline xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentAccess_version.xsd:679]: LengthType?

File: xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentTicketing_version.xsd [hunk 11]
```diff
@@ -217,7 +217,7 @@ Rail transport, Roads and Road transport
 			<xsd:element name="TactileInterfaceAvailable" type="xsd:boolean" minOccurs="0">
 				<xsd:annotation>
 					<xsd:documentation>Indicates whether there is a tactile interface +v1.1.</xsd:documentation>
-				</xsd:annotation>
+        </xsd:annotation>
```
Comment (skinkie) [inline xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentTicketing_version.xsd:220]:
```suggestion
				</xsd:annotation>
```

File: xsd/netex_part_1/part1_ifopt/netex_ifopt_path_support.xsd [hunk 12]
```diff
@@ -255,6 +255,7 @@ Rail transport, Roads and Road transport
 		</xsd:restriction>
 	</xsd:simpleType>
 	<!-- ======================================================================= -->
+
```
Comment (skinkie) [inline xsd/netex_part_1/part1_ifopt/netex_ifopt_path_support.xsd:258]:
```suggestion
```

**Comment by skinkie** [comment]
@ue71603 no i am not done with that 'other work'... just enough frustration to see this as a nice activity...

File: xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentAccess_version.xsd [hunk 13]
```diff
@@ -2,7 +2,7 @@
 <!-- edited with XMLSpy v2010 rel. 3 sp1 (http://www.altova.com) by Nicholas Knowles (Kizoom) -->
 <xsd:schema xmlns="http://www.netex.org.uk/netex" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:netex="http://www.netex.org.uk/netex" targetNamespace="http://www.netex.org.uk/netex" elementFormDefault="qualified" attributeFormDefault="unqualified" version="1.2.2" id="netex_ifopt_equipmentAccess_version">
 	<xsd:include schemaLocation="../../netex_framework/netex_reusableComponents/netex_facility_support.xsd"/>
-	<xsd:include schemaLocation="netex_ifopt_path_support.xsd"/>
+	<!-- <xsd:include schemaLocation="netex_ifopt_path_support.xsd"/> -->
```
Comment (skinkie) [inline xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentAccess_version.xsd:5]: Is this really your intention?

Comment (ue71603) [reply xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentAccess_version.xsd]: no.. trying to find the problem with the non-deterministic shema.

**Comment by skinkie** [comment]
Are you able to run xmllint locally. Or are you validating with another tool?

File: xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentSigns_version.xsd (hunk 7 above)
Comment (ue71603) [reply xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentSigns_version.xsd]: no longer exists. was better in master

**Comment by skinkie** [comment]
@ue71603 maybe pull in master in this repo?

**Comment by ue71603** [comment]
@Aurige @skinkie  finally.... it didn't remain much. That's the problem with so old stuff.

**Review by ue71603 — APPROVED** [review]

**Comment by ue71603** [comment]
and as it shrunk to almost nothing, it is clear, why we did not miss it in EPIAP: @Aurige  if some update is still needed in EPIAP: Can you do it.

File: xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentAccess_version.xsd [hunk 14]
```diff
@@ -1222,14 +1233,19 @@ Rail transport, Roads and Road transport
...
 			</xsd:element>
+			<xsd:element name="SuitableForCycles" type="xsd:boolean" minOccurs="0">
+				<xsd:annotation>
+					<xsd:documentation>Whether equipment is suitable for cycles.</xsd:documentation>
+				</xsd:annotation>
+			</xsd:element>
 			<xsd:element name="AudioPassthroughIndicator" type="xsd:boolean" minOccurs="0">
 				<xsd:annotation>
-					<xsd:documentation>Whether there is an audio signal indicating passing through. +v1.1</xsd:documentation>
+					<xsd:documentation>Whether there is an audio signal indicating passing trhrough</xsd:documentation>
```
Comment (skinkie) [inline xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentAccess_version.xsd:1243]:
```suggestion
					<xsd:documentation>Whether there is an audio signal indicating passing through</xsd:documentation>
```

File: xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentAccess_version.xsd [hunk 15]
```diff
@@ -1222,14 +1233,19 @@ Rail transport, Roads and Road transport
...
 				</xsd:annotation>
 			</xsd:element>
-			<xsd:element name="SuitableForCycles" type="xsd:boolean" minOccurs="0">
+			<xsd:element name="OpeningNecessaryForce" type="NecessaryForceEnumeration" minOccurs="0">
```
Comment (skinkie) [inline xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentAccess_version.xsd:1246]: Can you check if this part is intended, SuitableForCycles is removed?

File: xsd/netex_part_1/part1_ifopt/netex_ifopt_path_version.xsd [hunk 16]
```diff
@@ -193,7 +193,7 @@ Rail transport, Roads and Road transport
 				<xsd:annotation>
 					<xsd:documentation>Indicates whether there are guiding strips. +v1.1</xsd:documentation>
 				</xsd:annotation>
-			</xsd:element>
+			</xsd:element>
```
Comment (skinkie) [inline xsd/netex_part_1/part1_ifopt/netex_ifopt_path_version.xsd:196]:
```suggestion
			</xsd:element>
```

File: xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentAccess_version.xsd (hunk 15 above)
Comment (ue71603) [reply xsd/netex_part_1/part1_ifopt/netex_ifopt_equipmentAccess_version.xsd:1246]: It is still there. It was duplicated. See line 1236. It is just strange behaviour by the diff.

**Review by skinkie — APPROVED** [review]

**Comment by skinkie** [comment]
Thanks @ue71603

**Pull request merged**